src/App.js

```diff
@@ -652,6 +652,28 @@ export class App extends React.Component<AppProps, AppState> {
                                 return { announcementsEnabled: !(currentState.announcementsEnabled) };
                             });
                             break;
+                        case("addCommand"): {
+                            if (this.state.selectedAction) {
+                                const currentElement = document.activeElement;
+                                // $FlowFixMe: Not all elements have dataset property
+                                if (currentElement.dataset.controltype === 'programStep') {
```

Comment:
It seems like this could avoid a lot of duplicated code by making a single check for the allowed control types and a ternary operator for the index variable.

Reply:
Thanks for the suggestion, that's a good idea.